stock_force_assign_by_type/views/stock_picking_views.xml

```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="view_picking_form_force_assign" model="ir.ui.view">
+        <field name="model">stock.picking</field>
+        <field name="inherit_id" ref="stock.view_picking_form" />
+        <field name="arch" type="xml">
+            <field name="use_create_lots" position="after">
+                <field name="picking_type_force_reservation" />
+            </field>
+            <button name="do_unreserve" position="attributes">
+                <attribute
+                    name="attrs"
+                >{'invisible': ['|', '|', '|', '&amp;', ('picking_type_force_reservation', '=', False), ('picking_type_code', '=', 'incoming'), ('immediate_transfer', '=', True), '&amp;', ('state', '!=', 'assigned'), ('move_type', '!=', 'one'), '&amp;', ('state', 'not in', ('assigned', 'confirmed')), ('move_type', '=', 'one')]}</attribute>
+            </button>
+            <field name="forecast_availability" position="attributes">
+                <attribute
+                    name="attrs"
+                >{'column_invisible': ['|', '|', ('parent.state', 'in', ['draft', 'done']), '&amp;', ('parent.picking_type_force_reservation', '=', False), ('parent.picking_type_code', '!=', 'outgoing'), ('parent.immediate_transfer', '=', True)]}</attribute>
+            </field>
```

Comment:
Maybe it would be better to use: https://github.com/OCA/server-tools/tree/15.0/base_view_inheritance_extension

What do you think @pedrobaeza ?

Reply:
AFAIK, the module doesn't support `attrs`. It's been partially added in https://github.com/OCA/server-tools/pull/2824 if I read it well.

Reply:
Oh, ok. Let's put a good warning comment then

Reply:
done!